src/main.rs

```diff
@@ -23,27 +26,76 @@ use ump_stream::UmpTransformStream;
 #[global_allocator]
 static GLOBAL: mimalloc::MiMalloc = mimalloc::MiMalloc;
 
+fn try_get_fd_listeners() -> (Option<UnixListener>, Option<TcpListener>) {
+    let mut fd = ListenFd::from_env();
+
+    let unix_listener = if let Ok(fd_unix) = env::var("FD_UNIX") {
+        let fd_pos = fd_unix.parse().expect("FD_UNIX is not a number");
+        println!("Trying to take Unix socket at position {}", fd_pos);
+        Some(
+            fd.take_unix_listener(fd_pos)
+                .expect(format!("fd {} is not a Unix socket", fd_pos).as_str())
+                .expect(format!("fd {} has already been used", fd_pos).as_str()),
+        )
+    } else {
+        None
+    };
+
+    let tcp_listener = if let Ok(fd_tcp) = env::var("FD_TCP") {
+        let fd_pos = fd_tcp.parse().expect("FD_TCP is not a number");
+        println!("Trying to take TCP listener at position {}", fd_pos);
+        Some(
+            fd.take_tcp_listener(fd_pos)
+                .expect(format!("fd {} is not a TCP listener", fd_pos).as_str())
+                .expect(format!("fd {} has already been used", fd_pos).as_str()),
+        )
+    } else {
+        None
+    };
```

Comment:
```suggestion
    let tcp_listener = env::var("FD_TCP").ok().map(|fd_tcp| {
        let fd_pos = fd_tcp.parse().expect("FD_TCP is not a number");
        println!("Trying to take TCP listener at position {}", fd_pos);
        fd.take_tcp_listener(fd_pos)
            .expect(format!("fd {} is not a TCP listener", fd_pos).as_str())
            .expect(format!("fd {} has already been used", fd_pos).as_str())
    });
```
could be done in case